app/services/search/search_builder.rb

```diff
@@ -49,7 +48,9 @@ def build_search_replica
   end
 
   def build_suggestions
-    @wider_radius_suggestions = Search::SuggestionsBuilder.new(search_params, @params_hash[:radius]).radius_suggestions
+    return @wider_search_suggestions = Search::RadiusSuggestionsBuilder.new(search_params, @params_hash[:radius]).radius_suggestions if @point_coordinates.present?
+
+    @wider_search_suggestions = Search::BufferSuggestionsBuilder.new(@params_hash[:location], search_params).buffer_suggestions if @location_search.location_category_search?
```

Comment:
It would be safer to check whether the @location_search has a polygon.

`location_category_search?` just checks whether it's the kind of search that *ought* to have a polygon. But knowing us, we might have lost the polygon in transit.

Reply:
`if @location_search.location_polygon.present?`

Reply:
I find it a little hard to read with the `if`s far off to the right. Might be worth going

```ruby
if x
  do y
elsif z
  do a
end
```